diff --git a/src/sysfiles.c b/src/sysfiles.c
index 04731c8e8..37438fe71 100644
--- a/src/sysfiles.c
+++ b/src/sysfiles.c
@@ -92,12 +92,17 @@ extern int errno;

 /* utility to check return value of 'write'  */
 ssize_t writeandcheck(int fd, const char *buf, size_t count) {
-  int ret;
-  ret = write(fd, buf, count);
-  if (ret < 0)
-    ErrorQuit("Cannot write to file descriptor %d, see 'LastSystemError();'\n",
-               fd, 0L);
-  return ret;
+    static int depth = 0; /* Recursion depth */
+    int ret;
+    ret = write(fd, buf, count);
+    if (ret < 0 && depth == 0)
+    {
+        depth++;
+        ErrorQuit("Cannot write to file descriptor %d, see 'LastSystemError();'\n",
+                   fd, 0L);
+        depth--;
+    }
+    return ret;
 }


