commit b7d0dfaf1a8fe33f555679ebaf132caa4d9842c9
Author: Sebastian Reuße <seb@wirrsal.net>
Date:   Wed May 24 21:35:51 2017 +0200

ipc-unix: don’t truncate the message on EAGAIN

Fixes #4452.

diff --git a/input/ipc-unix.c b/input/ipc-unix.c
index e4b98edfc..f26e0cadd 100644
--- a/input/ipc-unix.c
+++ b/input/ipc-unix.c
@@ -78,12 +78,9 @@ static int ipc_write_str(struct client_arg *client, const char *buf)
                 return 0;
             }

-            if (errno == EINTR)
+            if (errno == EINTR || errno == EAGAIN)
                 continue;

-            if (errno == EAGAIN)
-                return 0;
-
             return rc;
         }

