diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ef2243..a617e13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
 
+option(WITH_GTK2_PREVIEW "Build with GTK+ 2 preview support" ON)
+option(WITH_GTK3_PREVIEW "Build with GTK+ 3 preview support" ON)
+
 add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories(
     ${CMAKE_SOURCE_DIR} 
@@ -45,7 +48,13 @@ install(TARGETS kcm_cgc DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES ${GHNS_FILES} DESTINATION ${CONFIG_INSTALL_DIR})
 install(FILES kde-gtk-config.desktop DESTINATION ${SERVICES_INSTALL_DIR})
 
-add_subdirectory(gtkproxies)
-add_subdirectory(gtk3proxies)
+if (WITH_GTK2_PREVIEW)
+    add_subdirectory(gtkproxies)
+endif (WITH_GTK2_PREVIEW)
+
+if (WITH_GTK3_PREVIEW)
+    add_subdirectory(gtk3proxies)
+endif (WITH_GTK3_PREVIEW)
+
 add_subdirectory(icons)
 add_subdirectory(tests)
