https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=686152

diff --git a/hw/xfree86/loader/loadmod.c b/hw/xfree86/loader/loadmod.c
index 73dc1b8..abf9b19 100644
--- a/hw/xfree86/loader/loadmod.c
+++ b/hw/xfree86/loader/loadmod.c
@@ -1099,19 +1099,18 @@ RemoveChild(ModuleDescPtr child)
     parent = child->parent;
     if (parent->child == child) {
         parent->child = child->sib;
-        return;
     }
-
-    prevsib = parent->child;
-    mdp = prevsib->sib;
-    while (mdp && mdp != child) {
-        prevsib = mdp;
-        mdp = mdp->sib;
+    else {
+        prevsib = parent->child;
+        mdp = prevsib->sib;
+        while (mdp && mdp != child) {
+            prevsib = mdp;
+            mdp = mdp->sib;
+        }
+        if (mdp == child)
+            prevsib->sib = child->sib;
     }
-    if (mdp == child)
-        prevsib->sib = child->sib;
     child->sib = NULL;
-    return;
 }

 void
