commit aa4c2aeadab0f52354982a147caaf594c86ad758
Author: Ilya Tumaykin <itumaykin@gmail.com>
Date:   Wed Mar 16 19:43:58 2016 +0300

    cmake: purge remnants of opencc after b6d872bb

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 39f460c..ddd2e99 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -45,7 +45,7 @@ set (UCHARDET_TARGET libuchardet)
 set (UCHARDET_STATIC_TARGET libuchardet_static)

 add_definitions(
-	-DPKGDATADIR="${DIR_SHARE_OPENCC}"
+	-DPKGDATADIR="${DIR_SHARE_UCHARDET}"
 	-DLOCALEDIR="${DIR_SHARE_LOCALE}"
 	-DVERSION="${UCHARDET_VERSION}"
 	-DPACKAGE_NAME="${PACKAGE_NAME}"
