diff --git a/waftools/checks/custom.py b/waftools/checks/custom.py
index 698742406f..1286d45d80 100644
--- a/waftools/checks/custom.py
+++ b/waftools/checks/custom.py
@@ -112,3 +112,29 @@ def check_openal(ctx, dependency_identifier):
         if fn(ctx, dependency_identifier):
             return True
     return False
+
+
+# Gentoo-specific bits
+__all__ += ["check_rpi"]
+
+
+def check_rpi(ctx, dependency_identifier):
+    # We need MMAL/bcm_host/dispmanx APIs.
+    # Upstream keeps pkgconfig files in '/opt/vc/lib/pkgconfig'.
+    # See https://github.com/raspberrypi/userland/issues/245
+    # PKG_CONFIG_SYSROOT_DIR helps with cross compilation.
+    prev_pkg_path = os.getenv('PKG_CONFIG_PATH', '')
+    os.environ['PKG_CONFIG_PATH'] = os.pathsep.join(
+        filter(None, [os.path.join(os.getenv('PKG_CONFIG_SYSROOT_DIR', '/'),
+                                   'opt/vc/lib/pkgconfig'),
+                      prev_pkg_path]))
+
+    checks = [
+        check_pkg_config('bcm_host', uselib_store='bcm_host'),
+        check_cc(lib=['mmal_core', 'mmal_util', 'mmal_vc_client'], use=['bcm_host']),
+        check_pkg_config('egl'),
+    ]
+
+    ret = all((fn(ctx, dependency_identifier) for fn in checks))
+    os.environ['PKG_CONFIG_PATH'] = prev_pkg_path
+    return ret
diff --git a/wscript b/wscript
index 750ce3bd1f..360db8e33c 100644
--- a/wscript
+++ b/wscript
@@ -757,25 +757,9 @@ video_output_features = [
         'deps': 'win32-desktop && shaderc && crossc',
         'func': check_cc(header_name=['d3d11_1.h', 'dxgi1_2.h']),
     }, {
-        # We need MMAL/bcm_host/dispmanx APIs. Also, most RPI distros require
-        # every project to hardcode the paths to the include directories. Also,
-        # these headers are so broken that they spam tons of warnings by merely
-        # including them (compensate with -isystem and -fgnu89-inline).
         'name': '--rpi',
         'desc': 'Raspberry Pi support',
-        'func': compose_checks(
-            check_cc(cflags=["-isystem/opt/vc/include",
-                             "-isystem/opt/vc/include/interface/vcos/pthreads",
-                             "-isystem/opt/vc/include/interface/vmcs_host/linux",
-                             "-fgnu89-inline"],
-                     linkflags="-L/opt/vc/lib",
-                     header_name="bcm_host.h",
-                     lib=['mmal_core', 'mmal_util', 'mmal_vc_client', 'bcm_host']),
-            # We still need all OpenGL symbols, because the vo_opengl code is
-            # generic and supports anything from GLES2/OpenGL 2.1 to OpenGL 4 core.
-            check_cc(lib="EGL"),
-            check_cc(lib="GLESv2"),
-        ),
+        'func': check_rpi,
     } , {
         'name': '--ios-gl',
         'desc': 'iOS OpenGL ES hardware decoding interop support',
