commit b12b92822182fddd2fc3d5116d505371ac72719e
Author: wm4 <wm4@nowhere>
Date:   Sat Sep 30 14:19:34 2017 +0200

video: fix green shit

diff --git a/video/img_format.c b/video/img_format.c
index 802528a91b..39bb751ddb 100644
--- a/video/img_format.c
+++ b/video/img_format.c
@@ -129,7 +129,7 @@ struct mp_imgfmt_desc mp_imgfmt_get_desc(int mpfmt)
         fmt == AV_PIX_FMT_UYYVYY411)
         return mp_only_imgfmt_desc(mpfmt);
     enum mp_component_type is_uint =
-        mp_imgfmt_get_component_type(fmt) == MP_COMPONENT_TYPE_UINT;
+        mp_imgfmt_get_component_type(mpfmt) == MP_COMPONENT_TYPE_UINT;

     struct mp_imgfmt_desc desc = {
         .id = mpfmt,
