diff --git a/libaegisub/Makefile b/libaegisub/Makefile
index 4efb685..7443eac 100644
--- a/libaegisub/Makefile
+++ b/libaegisub/Makefile
@@ -44,7 +44,8 @@ aegisub_OBJ += $(d)common/dispatch.o
 endif
 
 aegisub_PCH := $(d)lagi_pre.h
-aegisub_CPPFLAGS := -I$(d)include -I$(TOP) $(CPPFLAGS_BOOST) $(CFLAGS_LUA)
+aegisub_CPPFLAGS := -I$(d)include -I$(TOP) \
+	$(CPPFLAGS_BOOST) $(CFLAGS_LUA) $(CFLAGS_PTHREAD)
 
 $(d)common/charset_conv.o_FLAGS := $(CFLAGS_ICONV)
 $(d)common/parser.o_FLAGS := -ftemplate-depth=256
diff --git a/tools/Makefile b/tools/Makefile
index d9f64b8..f9a6cd3 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -8,7 +8,8 @@ endif
 
 repack-thes-dict_OBJ  := $(d)repack-thes-dict.o $(TOP)lib/libaegisub.a
 repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
-repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include $(CFLAGS_ICU)
+repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include \
+	$(CFLAGS_ICU) $(CFLAGS_PTHREAD)
 
 PROGRAM += $(d)repack-thes-dict
 
