diff --git a/libaegisub/Makefile b/libaegisub/Makefile
index 64204ef..bdcdc24 100644
--- a/libaegisub/Makefile
+++ b/libaegisub/Makefile
@@ -41,7 +41,8 @@ aegisub_OBJ += $(d)common/dispatch.o
 endif
 
 aegisub_PCH := $(d)lagi_pre.h
-aegisub_CPPFLAGS := -I$(d)include -I$(TOP) -fPIC $(CPPFLAGS_BOOST) $(CFLAGS_LUA)
+aegisub_CPPFLAGS := -I$(d)include -I$(TOP) -fPIC \
+	$(CPPFLAGS_BOOST) $(CFLAGS_LUA) $(CFLAGS_PTHREAD)
 
 $(d)common/charset_conv.o_FLAGS := $(CFLAGS_ICONV)
 $(d)common/parser.o_FLAGS := -ftemplate-depth=256
diff --git a/tools/Makefile b/tools/Makefile
index 81bcef3..f9a6cd3 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -7,8 +7,9 @@ PROGRAM += $(d)osx-bundle-restart-helper
 endif
 
 repack-thes-dict_OBJ  := $(d)repack-thes-dict.o $(TOP)lib/libaegisub.a
-repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU)
-repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include $(CFLAGS_ICU)
+repack-thes-dict_LIBS := $(LIBS_BOOST) $(LIBS_ICU) $(LIBS_PTHREAD)
+repack-thes-dict_CPPFLAGS := -I$(TOP) -I$(TOP)libaegisub/include \
+	$(CFLAGS_ICU) $(CFLAGS_PTHREAD)
 
 PROGRAM += $(d)repack-thes-dict
 
